### PR TITLE
Add table of contents to README

### PR DESCRIPTION
**Pull request by adityamenon**

### CODE REVIEW

**Comment by elifoster** [comment]
"Contents" might be a be more accurate header, since not all of them are examples. If that were to be changed, the bottom sections at the bottom would need to be added. It also might be useful to add this to the news file too.

**Comment by adityamenon** [comment]
Done.

File: README.md [hunk 1]
```diff
@@ -23,7 +23,20 @@ to change the interface of your classes or to change any other Ruby code.
...
+## Contents
+
+* [The famous `String#blank?` method](#the-famous-stringblank-method)
+* [Simple Sidekiq-compatible server](#simple-sidekiq-compatible-server)
+* [Safe-conversions](#safe-conversions)
+* [True parallelism](#true-parallelism)
+* [Defining-a-new-class](#defining-a-new-class)
+* [Replacing only several methods instead of the whole class](#replacing-only-several-methods-instead-of-the-whole-class)
+* [Class definition DSL](#class-definition-dsl)
+* [Calling Ruby code from Rust](#calling-ruby-code-from-rust)
```
Comment (malept) [inline README.md]: I would say everything above this should be under "Examples", an h2 header right below this list, and the bullet points above should be indented.

Comment (d-unseductable) [inline README.md]: @adityamenon, I agree with @malept. Can you please change this?

File: README.md [hunk 2]
```diff
@@ -23,7 +23,22 @@ to change the interface of your classes or to change any other Ruby code.
 
 As simple as Ruby, as efficient as Rust.
 
-## Examples
+## Contents
+
+* ### Examples
```
Comment (malept) [inline README.md]: This renders weirdly. Replace the header with a link.

File: README.md [hunk 3]
```diff
@@ -23,7 +23,22 @@ to change the interface of your classes or to change any other Ruby code.
...
+  - [Defining-a-new-class](#defining-a-new-class)
+  - [Replacing only several methods instead of the whole class](#replacing-only-several-methods-instead-of-the-whole-class)
+  - [Class definition DSL](#class-definition-dsl)
+  - [Calling Ruby code from Rust](#calling-ruby-code-from-rust)
+* [... and why is FFI not enough?](#-and-why-is-ffi-not-enough)
+* [How do I use it?](#how-do-i-use-it)
+* [Contributors are welcome!](#contributors-are-welcome)
+* [License](#license)
 
 ### The famous `String#blank?` method
```
Comment (malept) [inline README.md]: Add the Examples header above this line.

File: README.md [hunk 4]
```diff
@@ -23,8 +23,23 @@ to change the interface of your classes or to change any other Ruby code.
...
+  - [Replacing only several methods instead of the whole class](#replacing-only-several-methods-instead-of-the-whole-class)
+  - [Class definition DSL](#class-definition-dsl)
+  - [Calling Ruby code from Rust](#calling-ruby-code-from-rust)
+* [... and why is FFI not enough?](#-and-why-is-ffi-not-enough)
+* [How do I use it?](#how-do-i-use-it)
+* [Contributors are welcome!](#contributors-are-welcome)
+* [License](#license)
 
+## Examples
 ### The famous `String#blank?` method
```
Comment (malept) [inline README.md]: blank line between the h2 and the h3

**Comment by d-unseductable** [comment]
Thank you!

**Pull request merged**